sdk/include/reactos/shlobj_undoc.h

```diff
@@ -678,6 +678,34 @@ DECLARE_INTERFACE_(IShellBrowserService, IUnknown)
 #define IShellBrowserService_GetPropertyBag(T,a,b,c) (T)->lpVtbl->GetPropertyBag(T,a,b,c)
 #endif
 
+/*****************************************************************************
+ * IMruPidlList interface
+ */
+#define INTERFACE IMruPidlList
+DECLARE_INTERFACE_(IMruPidlList, IUnknown)
+{
+    /*** IUnknown ***/
+    STDMETHOD(QueryInterface)(THIS_ REFIID,PVOID*) PURE;
+    STDMETHOD_(ULONG,AddRef)(THIS) PURE;
+    STDMETHOD_(ULONG,Release)(THIS) PURE;
+    /*** IMruPidlList ***/
+    STDMETHOD(InitList)(THIS_ UINT, HKEY, LPCWSTR) PURE;
+    STDMETHOD(UsePidl)(THIS_ LPCITEMIDLIST, UINT*) PURE;
+    STDMETHOD(QueryPidl)(THIS_ LPCITEMIDLIST, UINT, UINT*, UINT*) PURE;
+    STDMETHOD(PruneKids)(THIS_ LPCITEMIDLIST) PURE;
+};
+#undef INTERFACE
+
+#ifdef COBJMACROS
+#define IMruPidlList_QueryInterface(T,a,b) (T)->lpVtbl->QueryInterface(T,a,b)
+#define IMruPidlList_AddRef(T) (T)->lpVtbl->AddRef(T)
+#define IMruPidlList_Release(T) (T)->lpVtbl->Release(T)
+#define IMruPidlList_InitList(T,a,b,c) (T)->lpVtbl->InitList(T,a,b,c)
+#define IMruPidlList_UsePidl(T,a,b) (T)->lpVtbl->UsePidl(T,a,b)
+#define IMruPidlList_QueryPidl(T,a,b,c,d) (T)->lpVtbl->QueryPidl(T,a,b,c,d)
+#define IMruPidlList_PruneKids(T,a) (T)->lpVtbl->PruneKids(T,a)
```

Comment:
I would suggest doing something to improve readability, like
```suggestion
#define IMruPidlList_QueryInterface(T,a,b)  (T)->lpVtbl->QueryInterface(T,a,b)
#define IMruPidlList_AddRef(T)              (T)->lpVtbl->AddRef(T)
#define IMruPidlList_Release(T)             (T)->lpVtbl->Release(T)
#define IMruPidlList_InitList(T,a,b,c)      (T)->lpVtbl->InitList(T,a,b,c)
#define IMruPidlList_UsePidl(T,a,b)         (T)->lpVtbl->UsePidl(T,a,b)
#define IMruPidlList_QueryPidl(T,a,b,c,d)   (T)->lpVtbl->QueryPidl(T,a,b,c,d)
#define IMruPidlList_PruneKids(T,a)         (T)->lpVtbl->PruneKids(T,a)
```